src/Evolution/Systems/ScalarTensor/StressEnergy.hpp

```diff
@@ -3,4 +3,93 @@
 
 #pragma once
 
-namespace ScalarTensor {}  // namespace ScalarTensor
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/CurvedScalarWave/Tags.hpp"
+#include "Evolution/Systems/ScalarTensor/Tags.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace ScalarTensor {
+
+/*!
+ * \brief Add in the trace-reversed stress-energy source term to the \f$\Pi\f$
+ * evolved variable of the ::gh system.
+ *
+ * \details The only stress energy source term in the Generalized Harmonic
+ * evolution equations is in the equation for \f$\Pi_{a b}\f$:
+ * \f[
+ * \partial_t \Pi_{ab} + \text{\{spatial derivative terms\}} =
+ * \text{\{GH source terms\}}
+ * - 16 \pi \alpha (T^{(\Psi)}_{ab} - \frac{1}{2} g_{a b} g^{cd}T^{(\Psi)}_{cd})
+ * ~.
+ * \f]
+ *
+ * (note that this function takes as argument the trace-reversed stress-energy
+ * tensor)
+ *
+ * This function adds that contribution to the existing value of `dt_pi`. The
+ * spacetime terms in the GH equation should be computed before passing the
+ * `dt_pi` to this function for updating.
+ *
+ * \see `gh::TimeDerivative` for details about the spacetime
+ * part of the time derivative calculation.
+ */
+void add_stress_energy_term_to_dt_pi(
```

Comment:
I was following the structure of GrMhdValencia. The idea is to call this function inside `TimeDerivative`, after computing the decoupled eqs for the metric and the scalar.

Reply:
Is this function going to become more complex at some point? Given that it is so simple it is probably  easier more readable to just write it out in `TimeDerivative` and add these docs to the action.

Reply:
See #5212

Reply:
yes, I gthink it would be more readable to just write out the function in the action. But if you prefer this, ok.

[optional] you can use tensor expressions here either way:
 
`tenex::evaluate(dt_pi, - 16. * M_PI * lapse() * trace_reversed_stress_energy(ti::a, ti::b);`

Reply:
Ok, I would prefer to leave this function. It also keeps the parallel structure with the grmhd system.